feat: Add option to choose image/video quality like WhatsApp #6316

JASIM0021 opened the pull request:
<!-- This is a pull request template, you do not need to uncomment or remove the comments, they won't show up in the PR text. -->

## Proposed changes
<!-- Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue below. -->
Added a button on the ShareView header to toggle between SD/HD. It can be just a text for now.

Default SD

## Issue(s)	
<!-- Link the issues being closed by or related to this PR. For example, you can use #594 if this PR closes issue number 594 -->
#6316
[NATIVE-849]

## How to test or reproduce
<!-- Mention how you would reproduce the bug if not mentioned on the issue page already. Also mention which screens are going to have the changes if applicable -->

## Screenshots
<img width="338" alt="Screenshot 2025-04-25 at 16 43 47" src="https://github.com/user-attachments/assets/a578fc67-9329-4b07-9309-0406e56d38ad" />

## Image with HD quality
![WhatsApp Image 2025-04-25 at 16 39 20](https://github.com/user-attachments/assets/6e17843a-fff3-4df6-baf5-00a3c6fc91d3)
##  Same Image with SD quality
![WhatsApp Image 2025-04-25 at 16 39 21](https://github.com/user-attachments/assets/87348552-fcfb-4760-af2d-bac5348c1a9e)

## video with HD quality 
![WhatsApp Image 2025-04-25 at 23 08 13](https://github.com/user-attachments/assets/5fa35901-22ff-4c80-bbe8-d740094319ff)

## Same video with SD quality 
![WhatsApp Image 2025-04-25 at 23 08 14](https://github.com/user-attachments/assets/8d1bab78-caa7-44a7-893c-7ba75c48baa9)

## Types of changes
<!-- What types of changes does your code introduce to Rocket.Chat? -->
<!-- Put an `x` in the boxes that apply -->

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist
<!-- Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code. -->

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [x] I have added necessary documentation (if applicable)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments
<!-- If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc... -->


[NATIVE-849]: https://rocketchat.atlassian.net/browse/NATIVE-849?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Added Media quality setting (SD/HD) in Settings, with explanatory labels.
  - Share screen now includes a quality toggle; SD compresses images/videos before sending to reduce size, HD sends original quality.
  - Settings section renamed to Media and reorganized to include the new quality options alongside auto-download preferences.
  - Updated in-app text for media settings to improve clarity (new titles, subtitles, and info for quality choices).
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

JASIM0021 committed:
feat: Add option to choose image/video quality like WhatsApp #6316

JASIM0021 committed:
using compresssor

diegolmello commented on app/views/ShareView/index.tsx:
```diff
@@ -42,6 +43,7 @@ interface IShareViewState {
 	loading: boolean;
 	readOnly: boolean;
 	attachments: IShareAttachment[];
+	compressionQuality: number;
```
Let's control the state like this and create constants for the quality values instead of hardcoding.

```
const QUALITY_SD = 0.5;
const QUALITY_HD = 1;
```

```suggestion
	quality: 'sd' | 'hd';
```

diegolmello commented on app/views/ShareView/index.tsx:
```diff
@@ -101,6 +104,15 @@ class ShareView extends Component<IShareViewProps, IShareViewState> {
+			const readOnly = await this.getReadOnly();
+			const { attachments, selected } = await this.getAttachments();
+			this.setState({ readOnly, attachments, selected }, () => this.setHeader());
+			this.startShareView();
```
Calling this is going to reset the composer. We should avoid calling it again.

JASIM0021 replied:
added by mistake(for testing purposes), but will remove them in the next commit.

diegolmello commented on app/views/ShareView/index.tsx:
```diff
@@ -101,6 +104,15 @@ class ShareView extends Component<IShareViewProps, IShareViewState> {
 		this.getServerInfo();
 	}
 
+	componentDidUpdate = async (_: Readonly<IShareViewProps>, prevState: Readonly<IShareViewState>, __?: any): Promise<void> => {
```
I don't get why this is necessary.

JASIM0021 replied:
removed

diegolmello commented on app/views/ShareView/index.tsx:
```diff
@@ -142,13 +154,30 @@ class ShareView extends Component<IShareViewProps, IShareViewState> {
+							title={this.state.compressionQuality === 0.5 ? 'SD' : 'HD'}
+							onPress={() => {
+								this.setState({ compressionQuality: this.state.compressionQuality === 0.5 ? 1 : 0.5 });
+								console.log('this.state.compressionQuality', this.state.compressionQuality);
```
Get rid of all console.logs

JASIM0021 replied:
done

diegolmello commented on app/views/ShareView/index.tsx:
```diff
@@ -237,11 +266,42 @@ class ShareView extends Component<IShareViewProps, IShareViewState> {
+		);
+		const processedAttachments = await Promise.all(
+			originalAttachments.map(async attachment => {
+				if (attachment.mime?.startsWith('image/')) {
```
Missing video

JASIM0021 replied:
video added

JASIM0021 committed:
video compression added

diegolmello commented on app/lib/constants/utils.ts:
```diff
@@ -0,0 +1,2 @@
+export const QUALITY_SD = 0.5;
```
Move it to ShareView.
Colocation strategy.

JASIM0021 replied:
Done

diegolmello commented on app.json:
```diff
@@ -1,3 +1,6 @@
 {
-	"name": "RocketChatRN"
+  "name": "RocketChatRN",
+  "plugins": [
+    "react-native-compressor"
+  ]
```
Why?

JASIM0021 replied:
It is required by  [react-native-compressor](https://www.npmjs.com/package/react-native-compressor) doc.

diegolmello replied:
> Add the Compressor plugin to your Expo config

We're not

diegolmello commented on app/views/ShareView/index.tsx:
```diff
@@ -36,12 +37,14 @@ import { sendFileMessage, sendMessage } from '../../lib/methods';
 	loading: boolean;
 	readOnly: boolean;
 	attachments: IShareAttachment[];
+	quality: 'sd' | 'hd';
```
```suggestion
	quality: 'SD' | 'HD';
```

JASIM0021 replied:
Done

diegolmello commented on app/views/ShareView/index.tsx:
```diff
@@ -142,13 +146,32 @@ class ShareView extends Component<IShareViewProps, IShareViewState> {
+				{!attachments.length && !readOnly ? (
+					<>
+						<HeaderButton.Item
+							title={this.state.quality.toUpperCase()}
```
Follow the existing code patterns and import quality on line 134.
Do it everywhere you're using `this.state.`

JASIM0021 replied:
Done

diegolmello commented on app/views/ShareView/index.tsx:
```diff
@@ -327,6 +355,82 @@ class ShareView extends Component<IShareViewProps, IShareViewState> {
+	/**
+	 * Processes attachments by compressing images while preserving other file types
+	 */
+	private processAttachments = async (originalAttachments: IShareAttachment[]): Promise<IShareAttachment[]> => {
```
```suggestion
	processAttachments = async (originalAttachments: IShareAttachment[]): Promise<IShareAttachment[]> => {
```

diegolmello commented on app/views/ShareView/index.tsx:
```diff
@@ -327,6 +355,82 @@ class ShareView extends Component<IShareViewProps, IShareViewState> {
 			this.messageComposerRef.current?.setInput(item.description || '');
 		}
 	};
+	/**
```
Since none of the compression methods deal with state management, you should remove them from `ShareView/index.tsx` and move them to `ShareView/utils.ts`.

diegolmello commented on app/views/ShareView/index.tsx:
```diff
@@ -327,6 +355,82 @@ class ShareView extends Component<IShareViewProps, IShareViewState> {
+	compressVideo = async (uri: string) => {
+		let outputOptions = {};
+
+		if (this.state.quality === 'sd') {
+			// SD compression options (lower resolution, bitrate, etc.) // please suggest for change
+			outputOptions = {
+				compressionMethod: 'manual', // or 'auto' depending on your requirement
+				bitrate: 1000000, // 1 Mbps for SD quality
+				maxSize: 5000000, // Max file size 5MB for SD (optional)
+				progressDivider: 10 // Adjust for how often you get progress updates
+			};
+		} else if (this.state.quality === 'hd') {
+			return uri;
+			// please suggest for any kind of change
+			// outputOptions = {
+			// 	compressionMethod: 'auto'
+			// 	// bitrate: 4000000, // 4 Mbps for HD quality
+			// 	// maxSize: 15000000,
+			// 	// progressDivider: 10 // Max file size 15MB for HD (optional)
+			// };
+		}
```
```suggestion
		if (quality === 'HD') {
			return uri;
		}

		const outputOptions = {
		  ...
		};
```

JASIM0021 replied:
Done

diegolmello commented on ios/Podfile.lock:
```diff
@@ -31,6 +31,9 @@ PODS:
     - SDWebImageAVIFCoder (~> 0.10.1)
     - SDWebImageSVGCoder (~> 1.7.0)
     - SDWebImageWebPCoder (~> 0.13.0)
+  - ExpoImageManipulator (11.8.0):
```
Run pods again.

JASIM0021 replied:
Done

JASIM0021 committed:
some changes

diegolmello committed:
Merge branch 'develop' into feat/add_option_to_image

diegolmello commented on ios/Podfile.lock:
```diff
@@ -169,9 +169,9 @@ PODS:
-  - MMKV (1.3.9):
-    - MMKVCore (~> 1.3.9)
-  - MMKVCore (1.3.9)
+  - MMKV (2.2.1):
+    - MMKVCore (~> 2.2.1)
+  - MMKVCore (2.2.1)
```
Unrelated to your PR

JASIM0021 replied:
I just ran pod install, and it was added automatically. I have no idea about this change and will remove it in the next commit.

diegolmello commented on app/views/ShareView/utils.ts:
```diff
@@ -0,0 +1,27 @@
+export const QUALITY_SD = 0.5;
+export const QUALITY_HD = 1;
+
+export const compressImage = async (uri: string, quality: string) => {
```
```suggestion
export const compressImage = async (uri: string, quality: 'SD' | 'HD') => {
```

diegolmello commented on app/views/ShareView/index.tsx:
```diff
@@ -327,6 +354,46 @@ class ShareView extends Component<IShareViewProps, IShareViewState> {
+	/**
+	 * Processes attachments by compressing images while preserving other file types
+	 */
+	processAttachments = async (originalAttachments: IShareAttachment[]): Promise<IShareAttachment[]> => {
```
If HD, no compression should run.

diegolmello commented on app/views/ShareView/index.tsx:
```diff
@@ -327,6 +354,46 @@ class ShareView extends Component<IShareViewProps, IShareViewState> {
+					let compressedPath = '';
+
+					if (attachment.mime?.startsWith('image/')) {
+						compressedPath = await compressImage(attachment.path, this.state.quality);
```
```suggestion
						compressedPath = await compressImage(attachment.path, quality);
```

diegolmello committed:
Make sure expo-image-picker doesn't use default compression, so react-native-compressor does it

diegolmello committed:
Rollback podfile.lock

diegolmello commented on app/views/ShareView/index.tsx:
```diff
@@ -36,12 +36,14 @@ import { sendFileMessage, sendMessage } from '../../lib/methods';
 	loading: boolean;
 	readOnly: boolean;
 	attachments: IShareAttachment[];
+	quality: 'SD' | 'HD';
```
```suggestion
   type TQuality = 'SD' | 'HD';

	quality: TQuality;
```

diegolmello commented on app/views/ShareView/utils.ts:
```diff
@@ -0,0 +1,27 @@
+	}
+
+	const result = await CompressVideo.compress(uri, {
+		compressionMethod: 'manual', // or 'auto' depending on your requirement
+		bitrate: 1000000, // 1 Mbps for SD quality
+		maxSize: 5000000, // Max file size 5MB for SD (optional)
+		progressDivider: 10 // Adjust for how often you get progress updates
```
Remove these comments

diegolmello commented on app/views/ShareView/utils.ts:
```diff
@@ -0,0 +1,27 @@
+import { Image as ImageCompressor, Video as CompressVideo } from 'react-native-compressor';
+
+export const QUALITY_SD = 0.5;
+export const QUALITY_HD = 1;
```
```suggestion

```

diegolmello commented on app/views/ShareView/index.tsx:
```diff
@@ -327,6 +354,46 @@ class ShareView extends Component<IShareViewProps, IShareViewState> {
+		}
+	};
+
+	// manipulateAsync(uri, [], { compress: quality, format: SaveFormat.JPEG });
```
Cleanup these comments

diegolmello commented:
@JASIM0021 Is it working when sharing from share extension?

JASIM0021 committed:
feat: Allow sending media in original quality on share extension

This change enables users to share media in its original, uncompressed quality via the share extension.  It also removes  plugin on app.json since it's not used anymore.

JASIM0021 committed:
Merge branch 'feat/add_option_to_image' of github.com:JASIM0021/Rocket.Chat.ReactNative into feat/add_option_to_image

JASIM0021 commented:
> @JASIM0021 Is it working when sharing from share extension?


Yes, it is working correctly when sharing from the share extension.

Here’s a demonstration:

https://github.com/user-attachments/assets/727795e0-2a86-47d8-a349-8cd492c8c0ca

diegolmello commented:
@JASIM0021 Awesome.
Make sure to do an output comparison of the same image uploaded both on SD and HD.
We should check size and quality.
Post a comment with it.

JASIM0021 commented:
> @JASIM0021 Awesome. Make sure to do an output comparison of the same image uploaded both on SD and HD. We should check size and quality. Post a comment with it.



> @JASIM0021 Awesome. Make sure to do an output comparison of the same image uploaded both on SD and HD. We should check size and quality. Post a comment with it.

## Using SD with share extension
![WhatsApp Image 2025-04-28 at 21 49 59](https://github.com/user-attachments/assets/339137fe-ff12-4e07-ac61-75871c3a9516)

## Using HD with share extension
![WhatsApp Image 2025-04-28 at 21 49 58](https://github.com/user-attachments/assets/2d7e813a-ab89-4fe4-945f-ac4a603e1645)

diegolmello commented:
@JASIM0021 Looking good!
We're just designing the SD/HD toggle icons and we're going to update your PR.
We've been also discussing if we should save the user choice on MMKV, so it's saved for future uploads.

We can take it from here.
It may not land on this month's release. Probably end of May.
Thanks for your contribution 👊🏼

JASIM0021 commented:
> @JASIM0021 Looking good! We're just designing the SD/HD toggle icons and we're going to update your PR. We've been also discussing if we should save the user choice on MMKV, so it's saved for future uploads.
> 
> We can take it from here. It may not land on this month's release. Probably end of May. Thanks for your contribution 👊🏼

Thanks for the update! 
Appreciate the heads-up on the timeline—glad to contribute!

– @diegolmello

diegolmello commented:
@JASIM0021 we've already made a decision regarding MMKV and I think you can do it to make sure this is merged faster.

- Rename this option to `Media` https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/f56fd1c9c0d01b723cf96d41916248f0f398c826/app/views/SettingsView/index.tsx#L217
- Add a title here https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/f56fd1c9c0d01b723cf96d41916248f0f398c826/app/views/MediaAutoDownloadView/index.tsx#L38
  - `<List.Section title='Media_auto_download'>` should do it
- Add a new List.Section above it for `Media quality`, similar to `Media auto download`
- The options are (List.Item title and subtitle)
  - `Standard`: Faster to send, smaller file size
  - `High definition`: Slower to send, can be a few times larger
- Add a List.Info with `Choose the quality of media files to be sent`
- Read useUserPreferences usage to understand how to implement this. It's very straightforward.

Thanks!

jeanfbrito commented:
![image](https://github.com/user-attachments/assets/4ae32597-fe4c-47e8-9222-5c766ad12582)

JASIM0021 commented:
@diegolmello  Thank you for your words! 🚀
I'm fully energised and on it. I'll make the necessary changes and push for updates as soon as possible.

JASIM0021 committed:
Merge branch 'develop' into feat/add_option_to_image

JASIM0021 committed:
feat: Add media quality settings

JASIM0021 commented:
@diegolmello  @jeanfbrito  Done ✅.
All requested changes have been implemented. Please review when you have time. Thanks!


https://github.com/user-attachments/assets/4dd04afc-ad28-4340-9dfa-accf3e137b17

JASIM0021 committed:
console remove type added

JASIM0021 committed:
section tite added

numandev1 commented on app/views/ShareView/utils.ts:
```diff
@@ -0,0 +1,27 @@
+export const QUALITY_SD = 0.5;
+
+export const compressImage = async (uri: string, quality: TQuality) => {
+	const result = await ImageCompressor.compress(uri, {
```
to get image and video quality similar to WhatsApp, you have to use `compressionMethod: 'auto' with maxWidth

JASIM0021 replied:
Done

JASIM0021 committed:
change compressionMethod manual to auto

diegolmello commented on app/views/ShareView/index.tsx:
```diff
@@ -142,13 +151,32 @@ class ShareView extends Component<IShareViewProps, IShareViewState> {
+						<HeaderButton.Item
+							title={quality}
+							onPress={() => {
+								this.setState({ quality: quality === 'SD' ? 'HD' : 'SD' }, () => {
```
```suggestion
								this.setState(({ quality }) => ({ quality: !quality })), () => {
```

JASIM0021 replied:
The type  flase is not  assignable to type of Tquality

diegolmello commented on app/views/ShareView/index.tsx:
```diff
@@ -237,11 +265,16 @@ class ShareView extends Component<IShareViewProps, IShareViewState> {
+		sendLoadingEvent({ visible: false });
+		const attachments = compressAttachment;
+
+		// return;
```
Remove

diegolmello commented on app/views/ShareView/index.tsx:
```diff
@@ -237,11 +265,16 @@ class ShareView extends Component<IShareViewProps, IShareViewState> {
-		const { attachments, room, text, thread, action, selected, selectedMessages } = this.state;
+		const { attachments: originalAttachments, room, text, thread, action, selected, selectedMessages, quality } = this.state;
+		sendLoadingEvent({ visible: true });
+		const compressAttachment = await this.processAttachments(originalAttachments, quality);
```
```suggestion
		const processedAttachment = await this.processAttachments(originalAttachments, quality);
```

diegolmello commented on app/views/ShareView/index.tsx:
```diff
@@ -327,16 +360,48 @@ class ShareView extends Component<IShareViewProps, IShareViewState> {
 			this.messageComposerRef.current?.setInput(item.description || '');
 		}
 	};
+	/**
+	 * Processes attachments by compressing images while preserving other file types
+	 */
```
Remove

diegolmello commented on app/views/ShareView/utils.ts:
```diff
@@ -0,0 +1,27 @@
+export type TQuality = 'SD' | 'HD';
+export const QUALITY_SD = 0.5;
+
+export const compressImage = async (uri: string, quality: TQuality) => {
```
You're only calling `compressImage` and `compressVideo` to compress them.
No point on receiving `quality` as a param.

```suggestion
export const compressImage = async (uri: string) => {
```

diegolmello commented on app/views/ShareView/utils.ts:
```diff
@@ -0,0 +1,27 @@
+export const compressImage = async (uri: string, quality: TQuality) => {
+	const result = await ImageCompressor.compress(uri, {
+		compressionMethod: 'auto',
+		maxWidth: 1000,
```
If possible, we should not change resolution.

JASIM0021 replied:
Yes, it's possible. I have made the change and tested it as well.

diegolmello commented on app/views/ShareView/utils.ts:
```diff
@@ -0,0 +1,27 @@
+	const result = await CompressVideo.compress(uri, {
+		compressionMethod: 'auto',
+		bitrate: 1000000,
+		maxSize: 5000000,
```
If possible, we should not limit size

diegolmello commented on app/views/ShareView/utils.ts:
```diff
@@ -0,0 +1,27 @@
+		compressionMethod: 'auto',
+		bitrate: 1000000,
+		maxSize: 5000000,
+		progressDivider: 10
```
Does it take time to compress? I think this is only needed when we need to render a progress bar?

JASIM0021 replied:
Yes, you are absolutely right; it's not necessary right now.

JASIM0021 committed:
feat: Implement media quality setting

This commit introduces a media quality setting, allowing users to choose between standard and high definition for media files. It also refactors code to use the  and  types from a single source.

diegolmello commented:
@JASIM0021 Going to focus back on this after the release, which should happen this Monday

JASIM0021 commented:
@diegolmello waiting for your review

Rohit3523 commented:
We recently updated the React Native version to 0.77.2, and the PR has already been merged into the dev branch. Could you please pull the latest changes and test if everything is working fine? Thanks!

Rohit3523 commented:
Also please resolve the conflicts, Thanks
<img width="850" alt="Screenshot 2025-05-11 at 2 16 28 PM" src="https://github.com/user-attachments/assets/a7f51e53-1bd1-486d-8054-066d8f21d63f" />

JASIM0021 committed:
Merge branch 'develop' of github.com:RocketChat/Rocket.Chat.ReactNative into feat/add_option_to_image

JASIM0021 committed:
upgrade compressor for v 77.2

JASIM0021 commented:
After upgrading to **React Native 0.77.2**, here’s a comparison of how media assets (images and videos) are rendered in **Standard Definition (SD)** vs **High Definition (HD)**.

---

##  Image Comparison

### SD vs HD Quality

| SD Quality                               | HD Quality                               |
| ---------------------------------------- | ---------------------------------------- |
| ![Image - SD](https://github.com/user-attachments/assets/c786b989-b2f4-4956-b7ca-bfc3afed2dd9) | ![Image - HD](https://github.com/user-attachments/assets/ec3d59fc-2dcd-4256-9e27-61fb72097e4e) |

---

## Video Thumbnail Comparison

### SD vs HD Quality

| SD Quality                               | HD Quality                               |
| ---------------------------------------- | ---------------------------------------- |
| ![Video - SD](https://github.com/user-attachments/assets/5d1ce6a9-ee64-4cba-9d49-0b629bdf4b3f) | ![Video - HD](https://github.com/user-attachments/assets/e3dba121-5361-46c4-8e2b-ec04660f2ab2) |

---

JASIM0021 committed:
Merge branch 'develop' into feat/add_option_to_image

JASIM0021 committed:
Merge branch 'develop' into feat/add_option_to_image

JASIM0021 committed:
Merge branch 'develop' of github.com:JASIM0021/Rocket.Chat.ReactNative into feat/add_option_to_image

JASIM0021 committed:
import marge fix

JASIM0021 committed:
Merge branch 'develop' into feat/add_option_to_image

JASIM0021 committed:
lint issue fix

JASIM0021 committed:
feat( typo) :- setMediaQualityPreference

JASIM0021 committed:
Merge branches 'feat/add_option_to_image' and 'feat/add_option_to_image' of github.com:JASIM0021/Rocket.Chat.ReactNative into feat/add_option_to_image

JASIM0021 committed:
feat(refactor):-  setMediaPref

JASIM0021 committed:
Update app/views/ShareView/index.tsx

JASIM0021 committed:
Update app/views/ShareView/index.tsx

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

JASIM0021 committed:
Update app/views/ShareView/utils.ts

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

JASIM0021 committed:
Update app/views/ShareView/utils.ts

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

diegolmello reviewed (commented):
Hey. It'll take a little bit longer for us to properly review.
PRs like this one are hard for the community to finish themselves alone, since it misses icons that are built internally (for example), so we'll have to finish ourselves after the PR reaches around 80% readiness, which is the case.

Take it as approved, but it'll take a bit to get merged.
Thanks for your contribution!
